File: Assets/Bundles/Client/Resources/TS/Controllers/Global/Camera/CameraTransform.lua
-- Compiled with unity-ts v2.1.0-75
--[[
	*
	* Unique class for holding the position and rotation of the camera. Used by `CameraMode` classes.
]]
local CameraTransform
do
	CameraTransform = setmetatable({}, {
		__tostring = function()
			return "CameraTransform"
		end,
	})
	CameraTransform.__index = CameraTransform
	function CameraTransform.new(...)
		local self = setmetatable({}, CameraTransform)
		return self:constructor(...) or self
	end
	function CameraTransform:constructor(position, rotation)
		self.position = if position ~= nil then position else Vector3.zero
		self.rotation = if rotation ~= nil then rotation else Quaternion.identity
	end
	function CameraTransform:fromTransform(transform)
		return CameraTransform.new(transform.position, transform.rotation)
	end
	function CameraTransform:Lerp(other, alpha)
		local position = self.position:Lerp(other.position, alpha)
		local rotation = Quaternion.Slerp(self.rotation, other.rotation, alpha)
		return CameraTransform.new(position, rotation)
	end
	CameraTransform.identity = CameraTransform.new()
end
return {
	CameraTransform = CameraTransform,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Bundles/@Easy/Core/Core~/src/Client/Controllers/Camera/CameraTransform.ts
/**
 * Unique class for holding the position and rotation of the camera. Used by `CameraMode` classes.
 */
export class CameraTransform {
	public static readonly Identity = new CameraTransform();

	/** The position of the camera in 3D world space. */
	public readonly Position: Vector3;

	/** The rotation of the camera in 3D world space. */
	public readonly Rotation: Quaternion;

	public static FromTransform(transform: Transform) {
		return new CameraTransform(transform.position, transform.rotation);
	}

	constructor(position?: Vector3, rotation?: Quaternion) {
		this.Position = position !== undefined ? position : Vector3.zero;
		this.Rotation = rotation !== undefined ? rotation : Quaternion.identity;
	}

	public Lerp(other: CameraTransform, alpha: number): CameraTransform {
		const position = this.Position.Lerp(other.Position, alpha);
		const rotation = Quaternion.Slerp(this.Rotation, other.Rotation, alpha);
		return new CameraTransform(position, rotation);
	}
}
